### DISCUSSION TITLE
RevalidatePath refetches font, causing jitter

### DISCUSSION THREAD

**Discussion by codinginflow** [vercel/next.js#50782] [App Router]
In the gif below, the button clicks are handled via server actions. After executing the server action, I call `revalidatePath` to refresh the page. This causes a visible jitter in the UI. In the network tab, I can see that the next/Font is reloaded and if I disable the font, the jitter is gone.

Did anyone run into the same issue? I know server actions are alpha but revalidatePath should be stable.

![flowmazon revalidate glitch](https://github.com/vercel/next.js/assets/52977034/6869ff1a-e442-410e-9e76-6fbd60c75755)

**Comment by elliotgonzalez123** [comment]
I'm having the same issue.  It's been a problem since app router was introduced.  The previous way of doing mutations, by using a client component and calling router.refresh() produces the same jitter.  I've also seen it in a few videos where people are demoing the app router features.  I'm surprised no one else has mentioned it.

**Comment by codinginflow** [comment]
I just noticed that the jitter is only there in dev mode. After building the project, it's gone.

**Reply by joyzhong** [reply]
Wow great catch, thanks.

I was running into this same flickering issue when using `router.refresh()`.

Anyone have a solution for this? Just spent hours debugging/looking for workarounds. Luckily the jitter disappears after building the project somehow, but this is really a basic feature that should work in dev.

**Reply by mordechaim** [reply]
Static resources are heavily cached in production, but not on dev mode.

**Reply by tshddx** [reply]
The root cause is that a `router.refresh()` causes the browser to download your global layout.css file again (which, for some reason, has the current Unix timestamp in the URL's query param, so it wouldn't be cached even if they tried) and that causes the browser to download the font files again. The font files at least have a constant URL based on what looks like a content hash, so they could easily set response headers to get the browser to cache them, but alas there are no such response headers.

**Reply by mordechaim** [reply]
It seems to be addressed [here](https://github.com/vercel/next.js/commit/a2f4ca52e4d7052d68a265db4ba43179205d8fd1)

**Reply by hnykda** [reply]
Omg, thanks! You saved my precious hours...

**Reply by SSTPIERRE2** [reply]
It seems to work fine in one next v14.0.4 app I'm working on, but [here's a sandbox](https://codesandbox.io/p/devbox/upvotes-server-actions-g9wl73?file=%2Fapp%2Fcomponents%2FUpvotes%2FUpvotes.tsx%3A28%2C32&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522cltg06zyq00073b6kqp4w64s2%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522cltg06zyq00023b6kxjz8003d%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522cltg06zyq00043b6kl7ul8f44%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522cltg06zyq00063b6k789fcxfk%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522cltg06zyq00023b6kxjz8003d%2522%253A%257B%2522id%2522%253A%2522cltg06zyq00023b6kxjz8003d%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cltg2o5k200023b6qcg9g9vs0%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A28%252C%2522startColumn%2522%253A32%252C%2522endLineNumber%2522%253A28%252C%2522endColumn%2522%253A32%257D%255D%252C%2522filepath%2522%253A%2522%252Fapp%252Fcomponents%252FUpvotes%252FUpvotes.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522cltg2o5k200023b6qcg9g9vs0%2522%257D%252C%2522cltg06zyq00063b6k789fcxfk%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cltg06zyq00053b6kpq7s4jgh%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3000%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522cltg06zyq00063b6k789fcxfk%2522%252C%2522activeTabId%2522%253A%2522cltg06zyq00053b6kpq7s4jgh%2522%257D%252C%2522cltg06zyq00043b6kl7ul8f44%2522%253A%257B%2522id%2522%253A%2522cltg06zyq00043b6kl7ul8f44%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cltg06zyq00033b6kaho715wp%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%255D%252C%2522activeTabId%2522%253A%2522cltg06zyq00033b6kaho715wp%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D) on v14.1.0 where it still has the flicker for some reason

**Reply by YashTotale** [reply]
Yes, still facing this issue with next v14.1.4

**Reply by sam-mahonri** [reply]
I'm on Next 14.2.4, and the problem still occurs, however, only in Firefox, in the other chromium-based browsers this problem no longer occurs.

**Reply by 4lejandrito** [reply]
Next 14.2.5 and having the same issue. Only happens in dev mode and Firefox. Chrome does not have the issue.
